### PR TITLE
Adds a managed script for pruning orphaned blobs from the image registry

### PR DESCRIPTION
**Pull request by clcollins**
This adds a managed script for pruning orphaned blogs from the image registry.

Fixes [OHSS-21555](https://issues.redhat.com//browse/OHSS-21555)

Signed-off-by: Chris Collins <collins.christopher@gmail.com>

### CODE REVIEW

**Comment by clcollins** [comment]
/hold

Struggling with some RBAC

File: scripts/SREP/hard-prune-image-registry/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+file: script.sh
+name: hard-prune-image-registry
+shortDescription: Manually prune image registry
```
Comment (mjlshen) [inline scripts/SREP/hard-prune-image-registry/metadata.yaml:3]:
```suggestion
shortDescription: Hard pruning the image registry
```

File: scripts/SREP/hard-prune-image-registry/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+description: Manually prunes orphaned blobs in the image registry; this should be used sparingly
```
Comment (mjlshen) [inline scripts/SREP/hard-prune-image-registry/metadata.yaml:4]:
```suggestion
description: Hard prunes orphaned blobs in the image registry; this causes the image registry to enter read-only mode during the pruning and should be used only when the customer acknowledges this. https://docs.openshift.com/container-platform/4.13/applications/pruning-objects.html#pruning-hard-pruning-registry_pruning-objects
```

Doesn't have to be this exact wording, but I was thinking we can elaborate on "sparingly" and link to the docs behind this process

File: scripts/SREP/hard-prune-image-registry/script.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    echo "Adding the image-pruner role to the service account..."
+    oc adm policy add-cluster-role-to-user system:image-pruner -z "${SERVICE_ACCOUNT}" -n ${NAMESPACE}
+    
+    # Mark the image registry read-only
+    echo "Marking the image registry read-only..."
+    oc patch configs.imageregistry.operator.openshift.io/cluster -p '{"spec":{"readOnly":true}}' --type=merge
+    
+    # Delete the orphaned blobs (delete mode: `--prune=delete`)
+    # This command can be run manually with `--prune=check` to see what would be deleted:
+    # EG: oc -n ${NAMESPACE} exec pod/${REGISTRY_POD} -- /bin/sh -c 'REGISTRY_LOG_LEVEL=info /usr/bin/dockerregistry -prune=check'
```
Comment (mjlshen) [inline scripts/SREP/hard-prune-image-registry/script.sh:108]: Do we still want these comments? I'm thinking no

**Review by mjlshen — COMMENTED** [review]
LGTM, just some wording nits

File: scripts/SREP/hard-prune-image-registry/script.sh (hunk 3 above)
Comment (clcollins) [reply scripts/SREP/hard-prune-image-registry/script.sh:108]: I added those for folks who might be following up, and considered adding a flag for the check command.  Fair point though, will remove.

File: scripts/SREP/hard-prune-image-registry/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      - get
+      - list
+      - watch
+    - apiGroups:
+      - ""
+      - image.openshift.io
+      resources:
+      - imagestreams/status
+      verbs:
+      - update
```
Comment (cblecker) [inline scripts/SREP/hard-prune-image-registry/metadata.yaml:145]:
```suggestion
    - apiGroups:
      - rbac.authorization.k8s.io
      resources:
      - clusterrolebindings
      verbs:
      - bind
      resourceNames:
      - system:image-pruner
```

Comment (clcollins) [reply scripts/SREP/hard-prune-image-registry/metadata.yaml]: Man, I wish I'd known I could do that yesterday!  :D  Thanks!

**Review by mjlshen — COMMENTED** [review]
/lgtm

**Pull request merged**